ballerina-test/src/test/java/org/ballerinalang/distribution/test/BallerinaCommandTest.java

```diff
@@ -0,0 +1,105 @@
+        Assert.assertTrue(actualOutput.contains("Distributions available remotely:"));
+        Assert.assertTrue(actualOutput.contains("1.* channel"));
+        Assert.assertTrue(actualOutput.contains("1.0.0"));
+        Assert.assertTrue(actualOutput.contains("1.1.0"));
+        Assert.assertTrue(actualOutput.contains("1.2.0"));
+        Assert.assertTrue(actualOutput.contains("Swan Lake channel"));
+        Assert.assertTrue(actualOutput.contains("slp1"));
+        Assert.assertTrue(actualOutput.contains("slp8"));
+        // test bal dist pull and fetching dependencies
+        actualOutput = TestUtils.executeCommand(path + " dist pull 1.2.3");
+        Assert.assertTrue(actualOutput.contains("Fetching the '1.2.3' distribution from the remote server..."));
+        Assert.assertTrue(actualOutput.contains("Fetching the dependencies for '1.2.3' from the remote server..."));
+        Assert.assertTrue(actualOutput.contains("Downloading jdk8u202-b08-jre"));
+        Assert.assertTrue(actualOutput.contains("'1.2.3' successfully set as the active distribution"));
+        TestUtils.testInstallation(path, "1.2.3", "2020R1", TOOL_VERSION, "1.2.3");
+        // test bal dist update
+        actualOutput = TestUtils.executeCommand(path + " dist update");
+        Assert.assertTrue(actualOutput.contains("Fetching the latest patch distribution for 'jballerina-1.2.3' from the remote server..."));
+        Assert.assertTrue(actualOutput.contains("Successfully set the latest patch distribution"));
+        actualOutput = TestUtils.executeCommand(path + " dist update");
+        Assert.assertTrue(actualOutput.contains("is already the active distribution"));
+        // test bal dist use
+        actualOutput = TestUtils.executeCommand(path + " dist use 1.2.3");
+        Assert.assertTrue(actualOutput.contains("'1.2.3' successfully set as the active distribution"));
+        TestUtils.testInstallation(path, "1.2.3", "2020R1", TOOL_VERSION, "1.2.3");
+        actualOutput = TestUtils.executeCommand(path + " dist pull slp7");
+        Assert.assertTrue(actualOutput.contains("Fetching the 'slp7' distribution from the remote server..."));
+        Assert.assertTrue(actualOutput.contains("Fetching the dependencies for 'slp7' from the remote server..."));
+        Assert.assertTrue(actualOutput.contains("Downloading jdk-11.0.8+10-jre"));
+        Assert.assertTrue(actualOutput.contains("'slp7' successfully set as the active distribution"));
+        TestUtils.testInstallation(path, "swan-lake-preview7", "v2020-09-22", TOOL_VERSION, "Preview 7");
+        // test bal dist remove <version>
+        actualOutput = TestUtils.executeCommand(path + " dist remove slp7");
+        Assert.assertTrue(actualOutput.contains("The active Ballerina distribution cannot be removed"));
+        actualOutput = TestUtils.executeCommand(path + " dist remove 1.2.3");
+        Assert.assertTrue(actualOutput.contains("Distribution '1.2.3' successfully removed"));
```

Comment:
@suganyasuven after deleting we can add another additional assertion to its not listing in `bal dist list`